src/schedule-and-details/schedule-section/certificate-display-row/CertificateDisplayRow.jsx

```diff
@@ -0,0 +1,194 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import {
+  Dropdown, Form, Collapsible, Icon,
+} from '@edx/paragon';
+import { Info as InfoIcon } from '@edx/paragon/icons';
+
+import { getLabelById } from '../../../utils';
+import { DatepickerControl, DATEPICKER_TYPES } from '../datepicker-control';
+import messages from './messages';
+
+export const CERTIFICATE_DISPLAY_BEHAVIOR = {
+  earlyNoInfo: 'early_no_info',
+  end: 'end',
+  endWithDate: 'end_with_date',
+};
+
+const CertificateDisplayRow = ({
+  intl,
+  certificateAvailableDate,
+  availableDateErrorFeedback,
+  certificatesDisplayBehavior,
+  displayBehaviorErrorFeedback,
+  onChange,
+}) => {
+  const dropdownOptions = [
+    {
+      id: CERTIFICATE_DISPLAY_BEHAVIOR.earlyNoInfo,
+      label: intl.formatMessage(messages.certificateBehaviorDropdownOption1),
+    },
+    {
+      id: CERTIFICATE_DISPLAY_BEHAVIOR.end,
+      label: intl.formatMessage(messages.certificateBehaviorDropdownOption2),
+    },
+    {
+      id: CERTIFICATE_DISPLAY_BEHAVIOR.endWithDate,
+      label: intl.formatMessage(messages.certificateBehaviorDropdownOption3),
+    },
+  ];
+
+  const formatCertificateDisplayBehavior = (value) => {
+    if (!value) {
+      return '';
+    }
+
+    if (Object.values(CERTIFICATE_DISPLAY_BEHAVIOR).includes(value)) {
+      return value;
+    }
+
+    return value.split('.')[1].toLowerCase();
+  };
+
+  const handleOnChange = (optionId) => {
+    if (optionId !== CERTIFICATE_DISPLAY_BEHAVIOR.endWithDate) {
+      onChange('', 'certificateAvailableDate');
+    }
+    onChange(optionId, 'certificatesDisplayBehavior');
+  };
+
+  const formattedCertificateBehavior = formatCertificateDisplayBehavior(
+    certificatesDisplayBehavior,
+  );
+
+  const certificateDisplayValue = getLabelById(dropdownOptions, formattedCertificateBehavior)
+    || intl.formatMessage(messages.certificateBehaviorDropdownEmpty);
+
+  const showAvailableDate = formattedCertificateBehavior === CERTIFICATE_DISPLAY_BEHAVIOR.endWithDate;
+
+  const renderReadMoreToggle = () => {
```

Comment:
It may also be worth considering creating a separate file for this data (since it takes up a lot of space)

Reply:
I don't think this is the best idea I already have a `certificateDisplayRow` file for one responsibility